tools/cache_creator/units/cache_creator_tests.cpp

```diff
@@ -128,3 +129,82 @@ TEST_CASE("Full UUID roundtrip") {
   CHECK(cc::hexStringToUuid(hexStr, dumped));
   CHECK(dumped == uuid);
 }
+
+TEST_CASE("Get cache info from invalid PAL metadata blob") {
+  char badMetadata[10] = {0, 1, 2, 3, 4, 5, 6, 7, 8, 9};
+
+  llvm::StringRef noteBlob(badMetadata, sizeof(badMetadata));
+  llvm::Expected<cc::ElfLlpcCacheInfo> cacheInfoOrErr = cc::getCacheInfoFromMetadataBlob(noteBlob);
+
+  llvm::Error err = cacheInfoOrErr.takeError();
+  CHECK(err.isA<llvm::StringError>());
+
+  llvm::consumeError(std::move(err));
+}
+
+TEST_CASE("Get cache info from valid PAL metadata blob") {
+  const char *sampleMetadata = "---\n"
```

Comment:
nit: you can avoid having to write `"`" and `\n` after each line by using a raw string literal: https://en.cppreference.com/w/cpp/language/string_literal

Reply:
EG:
```
"a a\n"
"b b\n"
"c"
```
==>
```
R"(aa
bb
c)"
```